apps/api/src/domain/models/playlist/commands/add-audio-item-to-playlist/add-audio-item-to-playlist.command-handler.ts

```diff
@@ -0,0 +1,79 @@
+import { AggregateType, ResourceType } from '@coscrad/api-interfaces';
+import { CommandHandler } from '@coscrad/commands';
+import { Inject } from '@nestjs/common';
+import { Valid } from '../../../../../domain/domainModelValidators/Valid';
+import {
+    ID_MANAGER_TOKEN,
+    IIdManager,
+} from '../../../../../domain/interfaces/id-manager.interface';
+import { IRepositoryForAggregate } from '../../../../../domain/repositories/interfaces/repository-for-aggregate.interface';
+import { IRepositoryProvider } from '../../../../../domain/repositories/interfaces/repository-provider.interface';
+import { DeluxeInMemoryStore } from '../../../../../domain/types/DeluxeInMemoryStore';
+import { InMemorySnapshot } from '../../../../../domain/types/ResourceType';
+import { InternalError } from '../../../../../lib/errors/InternalError';
+import { REPOSITORY_PROVIDER_TOKEN } from '../../../../../persistence/constants/persistenceConstants';
+import { ResultOrError } from '../../../../../types/ResultOrError';
+import { BaseUpdateCommandHandler } from '../../../shared/command-handlers/base-update-command-handler';
+import { BaseEvent } from '../../../shared/events/base-event.entity';
+import { validAggregateOrThrow } from '../../../shared/functional';
+import { Playlist } from '../../entities';
+import { PlaylistItem } from '../../entities/playlist-item.entity';
+import { AddAudioItemToPlaylist } from './add-audio-item-to-playlist.command';
+import { AudioItemAddedToPlaylist } from './audio-item-added-to-playlist.event';
+
+@CommandHandler(AddAudioItemToPlaylist)
+export class AddAudioItemToPlaylistCommandHandler extends BaseUpdateCommandHandler<Playlist> {
+    protected aggregateType: AggregateType = AggregateType.playlist;
+
+    protected repositoryForCommandsTargetAggregate: IRepositoryForAggregate<Playlist>;
+
+    constructor(
+        @Inject(REPOSITORY_PROVIDER_TOKEN)
+        protected readonly repositoryProvider: IRepositoryProvider,
+        @Inject(ID_MANAGER_TOKEN) protected readonly idManager: IIdManager
+    ) {
+        super(repositoryProvider, idManager);
+
+        this.repositoryForCommandsTargetAggregate = repositoryProvider.forResource(
+            ResourceType.playlist
+        );
+    }
+
+    protected actOnInstance(
+        instance: Playlist,
+        { audioItemId }: AddAudioItemToPlaylist
+    ): ResultOrError<Playlist> {
+        return instance.addItem(
+            new PlaylistItem({
+                resourceCompositeIdentifier: { type: ResourceType.audioItem, id: audioItemId },
+            })
+        );
+    }
+
+    protected async fetchRequiredExternalState(
```

Comment:
This is more a note to myself. In the future, it may be possible to do this dynamically by using our `@RefersTo` decorator, i.e. the full COSCRAD data schema for the command or maybe the domain instance. Once we have update the instance, we could use the schema for Playlist to determine that `items` contains references to other resources and then validate that they exist. This would save writing these extra methods everywhere. If we could whittle down implementation of update commands to `actOnInstance`, that would be ideal.